src/main/java/at/ac/tuwien/kr/alpha/common/Head.java

```diff
@@ -0,0 +1,101 @@
+package at.ac.tuwien.kr.alpha.common;
+
+import at.ac.tuwien.kr.alpha.common.atoms.Atom;
+import at.ac.tuwien.kr.alpha.common.atoms.BasicAtom;
+import at.ac.tuwien.kr.alpha.common.atoms.Literal;
+import at.ac.tuwien.kr.alpha.common.terms.Term;
+
+import java.util.List;
+
+/**
+ * Represents the head of a rule, i.e., either a choice or a disjunction of atoms, but not both.
+ * For normal rules the disjunction contains only one atom.
+ *
+ * Copyright (c) 2017, the Alpha Team.
+ */
+public class Head {
```

Comment:
What about making `Head` abstract and adding two non-abstract subclasses `DisjunctiveHead` and `ChoiceHead`? I dislike the pattern of having two properties of which one needs to be empty/null.